mirrord/layer/src/connection.rs

```diff
@@ -1,130 +1,91 @@
-///
-/// There is support for connecting directly to an existing pod (instead of going through
-/// `KubernetesAPI::create`) by setting `LayerConfig::connect_tcp`.
-///
-/// This is used for mirrord tests only.
-///
-/// ## Operator
-///
-/// The [`OperatorApi`] takes over the connection procedure if `LayerConfig::operator` is
-/// set to `true` and an operator is installed in the cluster. We use `OperatorApi::discover`
-/// instead of [`KubernetesAPI`].
-pub(crate) async fn connect(
-    config: &LayerConfig,
-) -> (Sender<ClientMessage>, Receiver<DaemonMessage>) {
-    let progress = NoProgress;
-
-    if let Some(address) = &config.connect_tcp {
-        Connection(address)
-            .connect(&progress)
-            .await
-            .unwrap_or_else(|err| handle_error(err, config))
-    } else if config.operator && let Some(connection) = OperatorApi::discover(config).await.transpose() {
-        connection.unwrap_or_else(|err| handle_operator_error(err))
-    } else {
-        let k8s_api = KubernetesAPI::create(config)
-            .await
-            .unwrap_or_else(|err| handle_error(err, config));
-
-        let agent_ref = {
-            if let (Some(pod_agent_name), Some(agent_port)) =
-                (&config.connect_agent_name, config.connect_agent_port)
-            {
-                info!(
-                    "Reusing existing agent {:?}, port {:?}",
-                    pod_agent_name, agent_port
-                );
-                (pod_agent_name.to_owned(), agent_port)
-            } else {
-                let (pod_agent_name, agent_port) = tokio::time::timeout(
-                    Duration::from_secs(config.agent.startup_timeout),
-                    k8s_api.create_agent(&progress),
-                )
-                .await
-                .unwrap_or(Err(KubeApiError::AgentReadyTimeout))
-                .unwrap_or_else(|err| handle_error(err, config));
-                info!("Created new agent {pod_agent_name}, port {agent_port}");
-
-                // Set env var for children to re-use.
-                std::env::set_var("MIRRORD_CONNECT_AGENT", &pod_agent_name);
-                std::env::set_var("MIRRORD_CONNECT_PORT", agent_port.to_string());
+pub(crate) async fn connect(addr: SocketAddr) -> (Sender<ClientMessage>, Receiver<DaemonMessage>) {
+    let stream = match TcpStream::connect(addr).await {
+        Ok(stream) => stream,
+        Err(e) => {
+            error!("Couldn't connect to internal proxy: {e:?}");
+            graceful_exit!("{FAIL_STILL_STUCK:?}");
+        }
+    };
+    wrap_raw_connection(stream)
+}
 
-                (pod_agent_name, agent_port)
+fn wrap_raw_connection(
+    stream: TcpStream,
+) -> (mpsc::Sender<ClientMessage>, mpsc::Receiver<DaemonMessage>) {
+    let mut codec = actix_codec::Framed::new(stream, ClientCodec::new());
+
+    let (in_tx, mut in_rx) = mpsc::channel(CONNECTION_CHANNEL_SIZE);
+    let (out_tx, out_rx) = mpsc::channel(CONNECTION_CHANNEL_SIZE);
```

Comment:
Are `in_tx = ClientMessage` and `out_tx = DaemonMessage`?

```suggestion
    let (client_in_tx, mut client_in_rx) = mpsc::channel(CONNECTION_CHANNEL_SIZE);
    let (daemon_out_tx, daemon_out_rx) = mpsc::channel(CONNECTION_CHANNEL_SIZE);
```